pyhive/presto.py

```diff
@@ -104,7 +108,10 @@ def __init__(self, host, port='8080', username=None, catalog='hive', schema='def
         if protocol not in ('http', 'https'):
             raise ValueError("Protocol must be http/https, was {!r}".format(protocol))
         self._protocol = protocol
-
+        self._auth = None
+        if password:
+            self._auth = HTTPBasicAuth(username, self._password)
+            self._protocol = 'https'  # require https when authenticating
```

Comment:
Can you require this explicitly like https://github.com/dropbox/PyHive/blob/master/pyhive/hive.py#L88 and https://github.com/dropbox/PyHive/blob/master/pyhive/tests/test_hive.py#L175?

Someone might pass protocol=http and pass a password, and the protocol would get silently ignored.